client/src/Movies/Movie.js

```diff
@@ -37,15 +38,25 @@ export default class Movie extends Component {
   //   addToSavedList(this.state.movie)
   // }
 
+  componentDidUpdate = (prev) => {
+    const id = this.props.match.params.id;
+    if(id !== prev.match.params.id){
+      this.fetchMovie(id);
+    }
+  }
+
   render() {
     if (!this.state.movie) {
       return <div>Loading movie information...</div>;
     }
 
-    const { title, director, metascore, stars } = this.state.movie;
+    // const { title, director, metascore, stars } = this.state.movie;
+
     return (
       <div className="save-wrapper">
-        <div className="movie-card">
+        <MovieCard movie={this.state.movie} />
```

Comment:
Why not go ahead and destructure `movie` at the top of your `render` method since you use it multiple times?